source/adapters/native_cpu/program.cpp

```diff
@@ -40,6 +74,18 @@ UR_APIEXPORT ur_result_t UR_APICALL urProgramCreateWithBinary(
 
   auto hProgram = new ur_program_handle_t_(
       hContext, reinterpret_cast<const unsigned char *>(pBinary));
+  if (pProperties != nullptr) {
+    for (uint32_t i = 0; i < pProperties->count; i++) {
+      auto mdNode = pProperties->pMetadatas[i];
+      std::string mdName(mdNode.pName);
+      auto [Prefix, Tag] = splitMetadataName(mdName);
+      if (Tag == __SYCL_UR_PROGRAM_METADATA_TAG_REQD_WORK_GROUP_SIZE) {
+        native_cpu::ReqdWGSize_t reqdWGSize;
+        getReqdWGSize(mdNode, reqdWGSize);
```

Comment:
Checking the value returned by ` getReqdWGSize` seems to be missing (unless UR_ASSERT is implemented as throwing an exception)

Reply:
Thanks for spotting it, I've added a check